c10/cuda/CUDAStream.cpp

```diff
@@ -114,24 +120,25 @@ std::ostream& operator<<(std::ostream& stream, StreamIdType s) {
 // see Note [Hazard when concatenating signed integers]
 
 static inline StreamIdType streamIdType(StreamId s) {
-  int mask_for_type = (1 << kStreamTypeBits) - 1;
-  if (s && ((s & mask_for_type) == 0)) {
-    // Externally allocated streams have their id being the cudaStream_ptr
-    // so the bits corresponding to the type will be 0 and will collide with
-    // the default stream.
-    return StreamIdType::EXT;
+  // Externally allocated streams have their id being the cudaStream_ptr
+  // so the last bit will be 0
+  if (!(s & 1)) {
```

Comment:
How come you don't need to test s is nonzero anymore? Because IIRC the default stream is all zeros and special cased this way.

Reply:
I think the answer is, you've changed the internal representation of the overall StreamId, but it's a little hard for me to tell exactly what the change is, I have to reverse engineer it from the diff lol

Reply:
Yeah let me add comments

Reply:
Ugh, with this I cannot do 
```
We are obligated to treat the stream ID 0 as the default stream,
```
that is currently written in the comments, because for native streams I set last bit to 1